src/data_types/macros.rs

```diff
@@ -29,17 +40,29 @@ macro_rules! impl_anoncreds_object_identifier {
 
         impl Validatable for $i {
             fn validate(&self) -> Result<(), ValidationError> {
+                let legacy_regex = match stringify!($i) {
+                    "IssuerId" => &LEGACY_DID_IDENTIFIER,
+                    "CredentialDefinitionId" => &LEGACY_CRED_DEF_IDENTIFIER,
+                    "SchemaId" => &LEGACY_SCHEMA_IDENTIFIER,
+                    // TODO: we do not have correct validation for a revocation registry and definition id
+                    "RevocationRegistryId" => &LEGACY_DID_IDENTIFIER,
```

Comment:
Both are the same (rev reg id and rev reg def id). You can find regexes for this in identifiers.ts in afj